pkg/controller/cloud/node_controller.go

```diff
@@ -204,8 +211,16 @@ func (cnc *CloudNodeController) UpdateCloudNode(_, newObj interface{}) {
 		return
 	}
 
-	cloudTaint := getCloudTaint(node.Spec.Taints)
-	if cloudTaint == nil {
+	// Check if a node has been added to the initialized nodes map,
+	// meaning that it has already reached the end of the initializeNode method.
+	// If the name is found but the machine ID does not match then it is a new
+	// node using the same name as a node that was previously deleted.
+	cnc.initializedNodesMutex.Lock()
+	machineID, found := cnc.initializedNodes[node.Name]
+	cnc.initializedNodesMutex.Unlock()
+	alreadyInitialized := found && (machineID == node.Status.NodeInfo.MachineID)
```

Comment:
I believe `NodeInfo.MachineID` is only set if cAdvisor is enabled, not sure if we want to be relying on cAdvisor behavior in this way here 🤔

Reply:
Ah ok. Is there another way to uniquely identify a node like this?
Otherwise I think I can manage without it by removing a name from the map when it is eventually seen in an Update event where the node object appears initialized, and can be discarded for that reason from then on.